inst/markdown/nashConvergence-qttyTarget.Rmd

```diff
@@ -0,0 +1,327 @@
+                                 paste0("Not converged<br>",
+                                  "Deviation: ", round(100*.data$p47_implicitQttyTarget_dev_iter,2), "%", ifelse(p47_implicitQttyTarget_dev_iter >= 0, " > ", " < -"), round(100*cm_implicitQttyTarget_tolerance,2),"%"))
+         )
+  )
+
+g[["deviation"]] <- 
+  lapply(setNames(tgtRegions, tgtRegions), function(reg){
+    subGroup <- setNames(tgts %>% filter(ext_regi == reg) %>% pull(ttot), tgts %>% filter(ext_regi == reg) %>% pull(group))
+    lapply(subGroup, function(tgtPeriod){
+      d <- plotData %>% filter(ext_regi == reg, ttot == tgtPeriod)
+      p <- ggplot(d, aes_string(x = "iteration", y = "p47_implicitQttyTarget_dev_iter")) +
+        suppressWarnings(geom_rect(aes(xmin = .data$iteration - 0.5, xmax = .data$iteration + 0.5,ymin = cm_implicitQttyTarget_tolerance * -1,ymax = cm_implicitQttyTarget_tolerance, fill = converged, text = tooltip), alpha = 0.2)) +
+        suppressWarnings(geom_line(aes_string(group = 1, text = "tooltip"))) +
+        scale_y_continuous(breaks = c(-5*cm_implicitQttyTarget_tolerance, -cm_implicitQttyTarget_tolerance, 0, cm_implicitQttyTarget_tolerance, 5*cm_implicitQttyTarget_tolerance), labels = scales::percent_format(accuracy = 1)) +
+        scale_fill_manual(values = booleanColor) +
+        theme_bw() +
+        theme(strip.background = element_blank(),
+              axis.title.x=element_blank(),
+              legend.position = "none") + 
+        coord_cartesian(ylim=c(-5*cm_implicitQttyTarget_tolerance, +5*cm_implicitQttyTarget_tolerance)) +
+        ylab("Deviation (%)") +
+        scale_x_continuous(expand=c(0,0),breaks=1:max(d$iteration)) +
+        #xlim(0,max(d$iteration)+1) +
+        theme(plot.margin = margin(0,42,0,22, "pt")) # t,r,b,l
+        
+      plotly::ggplotly(p, tooltip = c("text"), height = 200) #, autosize = FALSE
+    })
+  })
+
+```
+
+
+
+## {.tabset}
+
+```{r, include=FALSE}
+
+# Creating sections
+out <- ""
+for(reg in tgtRegions){
+  out <- paste0(out,knitr::knit_expand(text=paste0("### ", reg, " {.tabset}\n\n")))
+  for(subGroup in tgts %>% filter(ext_regi == reg) %>% pull(group)){
+    
+    tgtPeriod <- strsplit(subGroup,"_",fixed=T)[[1]][1]
+    
+    out <- paste0(out,knitr::knit_expand(text=paste0("#### ", subGroup, "\n\n")))
+    
+    out <- paste0(out,knitr::knit_expand(text=paste0("##### Quantity and Tax\n\n")))
+    out <- paste0(out,"```{r , results = 'asis', fig.height=10}\n\n")
+    out <- paste0(out,"g$qtty_and_tax[['", reg, "']][['", subGroup, "']]\n\n")
+    out <- paste0(out,"```\n\n")
+    
+#    out <- paste0(out,knitr::knit_expand(text=paste0("##### Deviation per iteration\n\n")))
+    out <- paste0(out,"```{r , results = 'asis', fig.height=10}\n\n")
+    out <- paste0(out,"g$deviation[['", reg, "']][['", subGroup, "']]\n\n")
+    out <- paste0(out,"```\n\n")
+  }
+}
+
+cat(out,file="output.txt", append=FALSE)
```

Comment:
This currently writes to an output.txt file in the inst folder of the library. This is most likely not intended!?

Reply:
https://github.com/pik-piam/remind2/pull/741/commits/56600a8311bcf189122f57c1c988606cd70284eb